src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripScheduleWithOffset.java

```diff
@@ -43,6 +45,7 @@ public final class TripScheduleWithOffset implements TripSchedule {
 
     // Trip times are sorted based on the arrival times at stop 0,
     this.sortIndex = arrivalTimes.applyAsInt(0);
+    this.wheelchairBoarding = getOriginalTripTimes().getTrip().getWheelchairBoarding();
```

Comment:
Should we store this in an array/list in `TripPatternForDates`. This looks like it could have a big performance impact

Reply:
I've moved the expensive call into the method so it's only used when wheelchair accessibility is enabled. @t2gran has indicated that the structure will change during his refactoring.

Reply:
This line gets evaluated each time we create an `TripScheduleWithOffset`, which is each time we find a boardable trip, and will access memory outside the Raptor data types, even when not needing the information later on. My thought was to create an array in `TripPatternForDates `, and [when looping through the trips](https://github.com/opentripplanner/OpenTripPlanner/blob/ccbb3792b7f0f44553dde12deaaff421058022c9/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripPatternForDates.java#L94), adding the wheelchair boarding info to the array. Then in the access, we could just do an array lookup. What you've done now is ok too.

Reply:
Ok, I'm not 100 percent firm on what all those indices mean but did you mean something like this? https://github.com/leonardehrenfried/OpenTripPlanner/commit/6f47d848c35c1b68f9f0fe23877018815f6de3a3

Reply:
Exactly like that